### PR TITLE
Add sign contract API endpoint

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR contains following changes:

- extension of contract with PreparedAt date (this will be needed to compare it with SignedAt date with business rules that will be introduced in a separate branch) as required field
- extension of contract with SignedAt date as non required field
- extension of existing API endpoint for contract preparation to store the date
- creation of new API endpoint for contract signature (without business rules)

### CODE REVIEW

File: Src/Fitnet.IntegrationTests/Contracts/PrepareContract/PrepareContractRequestFaker.cs [hunk 1]
```diff
@@ -9,7 +9,8 @@ public PrepareContractRequestFaker(int minAge, int maxAge, int minHeight, int ma
         CustomInstantiator(faker =>
             new PrepareContractRequest(
                 faker.Random.Number(minAge, maxAge),
-                faker.Random.Number(minHeight, maxHeight)
+                faker.Random.Number(minHeight, maxHeight),
+                DateTimeOffset.Now
```
Comment (kamilbaczek) [inline Src/Fitnet.IntegrationTests/Contracts/PrepareContract/PrepareContractRequestFaker.cs:13]: We can use here method from faker faker.Date.RecentOffset()

File: Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractRequestFaker.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using SuperSimpleArchitecture.Fitnet.Contracts.SignContract;
+
+namespace SuperSimpleArchitecture.Fitnet.IntegrationTests.Contracts.SignContract;
+
+internal sealed class SignContractRequestFaker : Faker<SignContractRequest>
+{
+    public SignContractRequestFaker()
+    {
+        CustomInstantiator(_ =>
+            new SignContractRequest(DateTimeOffset.Now)
```
Comment (kamilbaczek) [inline Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractRequestFaker.cs:10]: Could you use new fake.Date.RecentOffset(); similar to example above?

File: Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractTests.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        // Act
+        var signContractResponse = await _applicationHttpClient.PatchAsJsonAsync(url, signContractRequest);
+
+        // Assert
+        signContractResponse.Should().HaveStatusCode(HttpStatusCode.NotFound);
+    }
+
+    private async Task<Guid> PrepareContract()
+    {
+        const int minAge = 18;
```
Comment (kamilbaczek) [inline Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractTests.cs:52]: Could we wrap those properties into record inside tests.

e.g internal record ContractParamters(...)
{
internal static ContractParamters CreateValid() => new (100,0,210);
}

Then we could reused it everywhere. What you think?

File: Src/Fitnet/Contracts/ContractsEndpoints.cs [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
 
 internal static class ContractsEndpoints
 {
-    public static void MapContracts(this IEndpointRouteBuilder app) => 
+    internal static void MapContracts(this IEndpointRouteBuilder app)
+    {
         app.MapPrepareContract();
+        app.MapSignContract();
+    }
+
```
Comment (kamilbaczek) [inline Src/Fitnet/Contracts/ContractsEndpoints.cs:14]: Could we remove enter

**Review by kamilbaczek — APPROVED** [review]
I suggested minor changes. If you think that they are useful please implement those. Overall good job!! I wait for next PR :D

File: Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractRequestFaker.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractRequestFaker.cs:10]: Done

File: Src/Fitnet.IntegrationTests/Contracts/PrepareContract/PrepareContractRequestFaker.cs (hunk 1 above)
Comment (meaboutsoftware) [reply Src/Fitnet.IntegrationTests/Contracts/PrepareContract/PrepareContractRequestFaker.cs:13]: Done

File: Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractTests.cs (hunk 3 above)
Comment (meaboutsoftware) [reply Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractTests.cs:52]: Done

File: Src/Fitnet/Contracts/ContractsEndpoints.cs (hunk 4 above)
Comment (meaboutsoftware) [reply Src/Fitnet/Contracts/ContractsEndpoints.cs]: Done

**Review by kamilbaczek — APPROVED** [review]

**Pull request merged**